packages/firestore/lite/src/api/reference.ts

```diff
@@ -1233,3 +1247,14 @@ export function newUserDataReader(
     serializer
   );
 }
+
+export function newExpUserDataWriter<T>(
```

Comment:
I'm surprised this doesn't live closer to the UserDataWriter. It doesn't seem like it belongs in `reference.ts`.

Also, it's confusing that this is named `newExpUserDataWriter` but it lives in the lite SDK?

Reply:
Huh, you do catch all the weirdness here :) I was hoping I could get away with this weird name. I ended up fixing this by applying your suggestion below. There is now a LiteUserDataWriter, an ExpUserDataWriter and an UserDataWriter (which we could name ClassicUserDataWriter, but I don't yet use this term in our class names). Note that LiteUserDataWriter and ExpUserDataWriter are essentially the same, but I kept them separate for clarity. I also re-exported the DocumentReference types from `/exp/src/api/reference` for additional clarity.